src/Storages/Statistics/UniqStatistic.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <Common/Arena.h>
+#include <Storages/Statistics/Statistics.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+
+namespace DB
+{
+
+class UniqStatistic : public IStatistic
+{
+    std::unique_ptr<Arena> arena;
+    AggregateFunctionPtr uniq_collector;
+    AggregateDataPtr data;
+    UInt64 result;
+public:
+    explicit UniqStatistic(const StatisticDescription & stat_, DataTypePtr data_type) : IStatistic(stat_), result(0)
+    {
+        arena = std::make_unique<Arena>();
+        AggregateFunctionProperties property;
+        property.returns_default_when_only_null = true;
+        uniq_collector = AggregateFunctionFactory::instance().get("uniq", NullsAction::IGNORE_NULLS, {data_type}, Array(), property);
```

Comment:
AFAIK `uniq` is based on BJKST and can grow quite significantly in terms of state size

![Memory-consumption-of-the-twelve-cardinality-estimation-algorithms-on-real-world-datasets](https://github.com/ClickHouse/ClickHouse/assets/9449405/9a47b0ac-f33a-4db0-8be0-2d92b90a7f30)

If you don't care about uniq statistics for cases, when it have > 2048 elements, uniqUpTo can be be more interesting alternative or some more state size friendly like uniqHLL or uniqCombined.

Reply:
I think `uniq` uses Flajolet-Martin sketch maintaining 65536 samples. It’s still a large number but I think it’s ok for now. Changing to another algorithm is easy :)

Reply:
Flajolet-Martin aka HLL does not maintain samples, it maintains the numbers of leading zero bits in hashes.